elasticsearch_dsl/mapping.py

```diff
@@ -69,11 +69,12 @@ def save(self, index, using='default'):
         else:
             analysis = self._collect_analysis()
             if analysis:
-                if es.cluster.state(index=index, metric='metadata')['metadata']['indices'][index]['state'] != 'close':
-                    # TODO: check if the analysis config is already there
-                    raise IllegalOperation(
-                        'You cannot update analysis configuration on an open index, you need to close index %s first.' % index)
-                es.indices.put_settings(index=index, body={'analysis': analysis})
+                existing_analysis = es.indices.get_settings(index)[index]['settings']['index'].get('analysis', {})
+                if analysis.items() > existing_analysis.items():
```

Comment:
I am wondering why the use of the `>` operator, shouldn't we just be looking for a subset?